src/components/ControlBar/ViewControlBar/DashboardsBar.js

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useRef, useEffect, createRef } from 'react'
+import { connect } from 'react-redux'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+
+import Content from './Content'
+import ShowMoreButton from './ShowMoreButton'
+import DragHandle from './DragHandle'
+import { getRowsFromHeight } from './controlBarDimensions'
+import { sGetControlBarUserRows } from '../../../reducers/controlBar'
+import { acSetControlBarUserRows } from '../../../actions/controlBar'
+import { apiPostControlBarRows } from '../../../api/controlBar'
+
+import classes from './styles/DashboardsBar.module.css'
+
+export const MIN_ROW_COUNT = 1
+export const MAX_ROW_COUNT = 10
+const rowClassMap = {
+    [MIN_ROW_COUNT]: 'one',
+    2: 'two',
+    3: 'three',
+    4: 'four',
+    5: 'five',
+    6: 'six',
+    7: 'seven',
+    8: 'eight',
+    9: 'nine',
+    [MAX_ROW_COUNT]: 'ten',
+}
+
+const DashboardsBar = ({
+    userRows,
+    updateUserRows,
+    expanded,
+    onExpandedChanged,
+}) => {
+    const [dragging, setDragging] = useState(false)
+    const userRowsChanged = useRef(false)
+    const ref = createRef()
+
+    const adjustRows = newHeight => {
+        const newRows = Math.max(
+            MIN_ROW_COUNT,
+            getRowsFromHeight(newHeight - 52) // don't rush the transition to a bigger row count
```

Comment:
What is `52` in this context?

Reply:
I'm not really sure. Planning to look into that, but this was code hasn't changed, so I didn't want to touch it right now.